lib/signet/sleuth.ex

```diff
@@ -30,13 +30,20 @@ defmodule Signet.Sleuth do
   defp query_internal(bytecode, query, selector, annotated, opts)
        when is_binary(bytecode) and is_list(opts) do
     {sleuth_address, opts} = Keyword.pop(opts, :sleuth_address, @sleuth_address)
-    {decode_binaries, rpc_opts} = Keyword.pop(opts, :decode_binaries, true)
+    {decode_binaries, opts} = Keyword.pop(opts, :decode_binaries, true)
+    {decode_structs, opts} = Keyword.pop(opts, :decode_structs, false)
+    {be_obvious, rpc_opts} = Keyword.pop(opts, :be_obvious, decode_structs)
 
     with {:ok, query_res_bytes} <-
            Signet.Contract.Sleuth.call_query(sleuth_address, bytecode, query, rpc_opts),
          {:ok, query_res} <- try_decode_bytes(query_res_bytes),
-         {:ok, res} <- try_decode(query_res, selector) do
-      {:ok, unwrap_outer_tuple(encode_map(res, selector.returns, annotated, decode_binaries))}
+         {:ok, res} <- try_decode(query_res, selector, decode_structs) do
+      {:ok,
+       postprocess(res, selector.returns,
+         annotated: annotated,
+         decode_binaries: decode_binaries,
+         be_obvious: be_obvious
+       )}
```

Comment:
i had to rewrite `encode_map` because it didn't work on the output of using `ABI.decode` with `decode_structs`.

i guess strictly-speaking i probably didn't have to rewrite it; i just needed to add a case that would handle `{:tuple, _}` types with map values instead of tuple values, but i didn't realize that until i had already rewritten it and discovered its various quirks, which led me to adding the `be_obvious` option since i didn't like all that.